addons/account/models/account_tax.py

```diff
@@ -1605,25 +1606,54 @@ def _round_base_lines_tax_details(self, base_lines, company, tax_lines=None):
             if not tax_amounts['base_lines']:
                 continue
 
-            base_line = tax_amounts['reference_base_line'] = max(
-                tax_amounts['base_lines'],
-                key=lambda base_line: base_line['tax_details']['total_included_currency'],
+            tax_amounts['sorted_base_line_x_tax_data'] = [
+                (
+                    base_line,
+                    next(
+                        (
+                            tax_data
+                            for tax_data in base_line['tax_details']['taxes_data']
+                            if tax_data['tax'] == tax and tax_data['is_reverse_charge'] == is_reverse_charge
+                        ),
+                        None,
+                    )
+                )
+                for base_line in sorted(
+                    tax_amounts['base_lines'],
+                    key=lambda base_line: -base_line['tax_details']['total_included_currency'],
+                )
+            ]
+            tax_amounts['total_included_currency'] = sum(
+                abs(base_line['tax_details']['total_included_currency'])
+                for base_line in tax_amounts['base_lines']
             )
-            if not tax:
+            if not tax or not tax_amounts['total_included_currency']:
                 continue
 
-            tax_details = base_line['tax_details']
             delta_tax_amount_currency = tax_amounts['raw_tax_amount_currency'] - tax_amounts['tax_amount_currency']
             delta_tax_amount = tax_amounts['raw_tax_amount'] - tax_amounts['tax_amount']
+            for delta, delta_field, delta_currency in (
+                (delta_tax_amount_currency, 'tax_amount_currency', currency),
+                (delta_tax_amount, 'tax_amount', company.currency_id),
+            ):
+                if delta_currency.is_zero(delta):
+                    continue
 
-            tax_data = next(
-                x
-                for x in tax_details['taxes_data']
-                if x['tax'] == tax and x['is_reverse_charge'] == is_reverse_charge
-            )
-            tax_amounts['reference_tax_data'] = tax_data
-            tax_data['tax_amount_currency'] += delta_tax_amount_currency
-            tax_data['tax_amount'] += delta_tax_amount
+                sign = -1 if delta < 0.0 else 1
+                nb_of_errors = round(abs(delta / delta_currency.rounding))
+                remaining_errors = nb_of_errors
+                for base_line, tax_data in tax_amounts['sorted_base_line_x_tax_data']:
+                    tax_details = base_line['tax_details']
+                    if not remaining_errors or not tax_data:
+                        break
+
+                    nb_of_amount_to_distribute = min(
+                        abs(math.ceil(tax_details['total_included_currency'] * nb_of_errors / tax_amounts['total_included_currency'])),
+                        remaining_errors,
+                    )
```

Comment:
```suggestion
                        math.ceil(abs(tax_details['total_included_currency'] * nb_of_errors / tax_amounts['total_included_currency'])),
                        remaining_errors,
                    )
```
The order of `math.ceil` and `abs` matters because math.ceil will round a negative number towards zero